### ISSUE TITLE
Normalizing number of insertions and reads per CDS using the number of insertions in local neutral DNA.

### ISSUE THREAD

**Issue by Gregory94** [Gregory94/LaanLab-SATAY-DataAnalysis#21]

**Comment by Gregory94** [comment]
In order to determine whether the number of transposon insertions in a CDS is more or less then what we would expect, the number of insertions in each CDS need to be normalized relative to the average number of transposon insertions outside the CDS (i.e. the background). However, the number of insertions is not homogeneously distributed throughout the genome (i.e. some genomic regions are more likely to have transposon insertions compared with other regions). Therefore, the number of insertions in the CDS need to be normalized by the average of the local background. Currently I'm developing a tool that determine this local background by searching, for each CDS, the local neutral regions (i.e. regions that do not seem to code for anything special like genes, telomeres, ARS etc.).

**Comment by Gregory94** [comment]
I made a first attempt in estimating the fitness of the cells based on number of reads. I went for the most simple approach which is dividing a chromosome in sections (which I called windows) and normalized the data within each window (see attached file). I did the normalization like this because the likelyhood of an insertion is not constant throughout chromosome, so some regions (typically around the centrosomes) typically bare more transposons. After I got a distribution that looked more flat (e.g. the average went from 0.69+/-1.53 reads per basepair per region before normalization to 0.66+/-1.33 after normalization). Finally, I divided all reads by the largest bar (assuming these cells had the highest fitness), which gives me a value between 0 and 1 which I then define as the fitness level.

I think this approach might be so good after all, since it is a bit dubious how to define the windows for normalization.
Also, there are typically outliers (i.e. few bars with much more reads compared to the other bars), even after normalization.
So I am going to look at other approaches as well.

[Fitness_from_reads1.pdf](https://github.com/Gregory94/LaanLab-SATAY-DataAnalysis/files/5244601/Fitness_from_reads1.pdf)

**Comment by leilaicruz** [comment]
Great Greg!
 The next step is to validate this fitness measure with the available information on positive and negative interactors of dpl1, using the correlation matrix of positive and negative scores (score=ab-a*b).

**Comment by Gregory94** [comment]
Normalization of the reads is updated with a method introduced in the paper by [Gallagher et.al. 2011](https://mbio.asm.org/content/2/1/e00315-10.short). This is done in [the code genomicfeatures_dataframe_with_normalization](https://github.com/Gregory94/LaanLab-SATAY-DataAnalysis/blob/dev_Gregory/Python_scripts/fitness_from%20SATAY_codes/genomicfeatures_dataframe_with_normalization.py). The normalization equation is given in the attached figure.
<img width="734" alt="Read_Normalization_Equation" src="https://user-images.githubusercontent.com/29129193/94900927-235f8480-0496-11eb-87f5-89b43633606f.PNG">
For the normalization of the interchromosomal regions, only the noncoding regions are considered. Attached are two plots of the same chromosome (chromosome V from WT1 from Benoit), the first which is not normalized and the second which is normalized using this code. 
![chrV_nonnorm](https://user-images.githubusercontent.com/29129193/94901892-a3d2b500-0497-11eb-895e-fc4b6dd6074c.png)
![chrV_norm](https://user-images.githubusercontent.com/29129193/94901955-be0c9300-0497-11eb-93c6-e85c5b57ed64.png)

This code creates a dataframe that contains all important information, the feature name and type (e.g. gene, telomeres, centromeres etc.), essentiality, genomic position, length, number of insertions, number of reads, normalized number of reads, number of insertions and reads in the central 80% of the genes, etc.

Currently it needs to be investigated how to determine a relevant window size and the codes needs to be properly tested and verified.

**Comment by leilaicruz** [comment]
To complete the fitness calculation we would also need the data from the normalized reads of the mutant : dpl1d

**Comment by leilaicruz** [comment]
- [x] Normalized reads from the mutant (https://github.com/Gregory94/LaanLab-SATAY-DataAnalysis/tree/dev_Leila/Python_scripts/fitness_from%20SATAY_codes)
- [ ] It will be nice to add the standard name of the feature name, so we can easily identify ORFs

**Comment by Gregory94** [comment]
I added the gene aliases in the dataframe dna_df2 (see [genomicfeatures_dataframe_with_normalization.py](https://github.com/Gregory94/LaanLab-SATAY-DataAnalysis/blob/dev_Gregory/Python_scripts/fitness_from%20SATAY_codes/genomicfeatures_dataframe_with_normalization.py)). This thus now includes the systematic gene names (e.g. YLR229C) and the standard name(s) (e.g. CDC42).

**Comment by Wteunisse** [comment]
An easy addition to this, as discussed in the meeting with Werner, is to divide the read number by the number of transposons, given that we have multiple transposons and reads added up as a result of the bin size or a moving average. This should help us to get rid of the peak around the centromere because this peak is only due to an increased number of transposons and not due to more reads per transposon.

**Comment by Wteunisse** [comment]
Leila commented here that we could check if this is useful by calculating the variation in the supposed neutral regions over the chromosome, as after this normalization we would expect little variation in the number of reads over all the neutral regions.

**Comment by Gregory94** [comment]
yes, I have already added this to the [dataframe variable](https://github.com/Gregory94/LaanLab-SATAY-DataAnalysis/blob/master/python_modules/genomicfeatures_dataframe_with_normalization.py) (see also [the python notebook](https://github.com/Gregory94/LaanLab-SATAY-DataAnalysis/blob/master/python_modules/genomicfeatures_dataframe_with_normalization.ipynb) of the same code) using the name `Nreadsperinsrt` and I also did this for the truncated gene where I ignore the first and last 100bp of the genes. I took the number of transposons per genomic feature (i.e. gene, telomere, noncoding region etc.). I haven't done any statistics on this data yet.

**Comment by Gregory94** [comment]
See also [this question](https://groups.google.com/forum/?oldui=1#!category-topic/satayusers/bioinformatics/PdktLyucbkE) at the satay forum about normalization.
